opennlp-tools/src/test/java/opennlp/tools/util/DownloadUtilTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package opennlp.tools.util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import opennlp.tools.sentdetect.SentenceModel;
+import opennlp.tools.tokenize.TokenizerModel;
+
+public class DownloadUtilTest {
+
+  @ParameterizedTest(name = "Verify \"{0}\" sentence model")
+  @ValueSource(strings = {"en", "fr", "de", "it", "nl"})
+  public void testDownloadModelByLanguage(String lang) throws IOException {
+    SentenceModel model = DownloadUtil.downloadModel(lang,
+            DownloadUtil.ModelType.SENTENCE_DETECTOR, SentenceModel.class);
```

Comment:
Is the test here actually downloading the model? If so, do you think we could mock it to run offline?

Reply:
It's actually downloading, like users would require it to download if they create XYZModel instances via the language code constructor.

Atm, I have no "good" idea how to mock the download in the case of the class unter test, that is DownloadUtil. However, I see your idea/point. Maybe worth a separate issue for (further) improvements?

Reply:
If the files are available at `System.getProperty("user.home") + "/.opennlp/")` and you are offline, the tests will pass. Otherwise if the tests were never run and you are offline, they will fail.

In order to mock it, we would need to adjust the `BASE_URL` part and inject a base url like `file://` pointing to a local location containing related (mock) models. Due to the `static` nature of `DownloadUtil`, this is pain ;-)

Side note: Having hard-coded values is a bit error prone (as shown by this issue ;-)). Perhaps it would be worth to parse the content of https://dlcdn.apache.org/opennlp/models/ud-models-1.0/ and obtain the model urls dynamically from that location (of course via separate issue).

Reply:
> If the files are available at System.getProperty("user.home") + "/.opennlp/") and you are offline, the tests will pass. Otherwise if the tests were never run and you are offline, they will fail.
> 

I can improve this part with the help of JUnit's `ExecutionCondition`. Stay tuned.

Reply:
+1 to @rzo1 suggestion to improve it, and also to not block this PR and move it to a follow-up issue. Thanks @mawiesne !

Reply:
> I can improve this part with the help of JUnit's `ExecutionCondition`. Stay tuned.

Done, pushed.